app/src/main/java/org/sopt/and/presentation/loginScreen/LoginViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package org.sopt.and.presentation.loginScreen
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.serialization.json.Json
+import org.sopt.and.data.datalocal.datasource.UserInfoLocalDataSource
+import org.sopt.and.domain.model.User
+import org.sopt.and.data.dataremote.model.request.RequestGetUserDto
+import org.sopt.and.data.dataremote.model.response.ResponseGetUserFailedDto
+import org.sopt.and.data.dataremote.network.ServicePool
+import org.sopt.and.domain.usecase.PostLoginUseCase
+import org.sopt.and.domain.usecase.SaveAccessTokenUseCase
+import org.sopt.and.domain.usecase.SaveUserNameUseCase
+import org.sopt.and.presentation.signupScreen.StringInputValidCheck
+import org.sopt.and.presentation.signupScreen.PasswordValidCheck
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val userInfoLocalDataSource: UserInfoLocalDataSource,
+    private val postLoginUseCase: PostLoginUseCase,
+    private val saveUserNameUseCase: SaveUserNameUseCase,
+    private val saveAccessTokenUseCase: SaveAccessTokenUseCase
+) : ViewModel() {
+
+    //로그인 요청을 서버로 보내기 위함
+    private val userService by lazy { ServicePool.userService }
+
+    private val _user = MutableStateFlow(User())
+    val user: StateFlow<User> = _user
+
+    private val _loginResult = MutableStateFlow<Boolean?>(null)
+    val loginResult = _loginResult.asStateFlow()
```

Comment:
```suggestion
    sealed class LoginUiState {
    object Initial : LoginUiState()
    object Success : LoginUiState()
    data class Error(val message: String) : LoginUiState()
}

private val _loginState = MutableStateFlow<LoginUiState>(LoginUiState.Initial)
val loginState = _loginState.asStateFlow()
```
Boolean? 대신 sealed class를 사용하여 상태를 더 명확하게 표현할 수 있어요!